### PR TITLE
fix: When adding a room param to urls check for previous params.

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

**Comment by damencho** [comment]
@saghul There is more mobile-related code that I have not changed ... but I think we need first to discuss how we approach this from the mobile side of things...

**Review by saghul — COMMENTED** [review]
Left some comments, PTAL. What specific scenario do we have that requires this change? Asking out of curiosity, but also because of the mobile app update cycle.

File: connection.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {
     JitsiConnectionEvents
 } from './react/features/base/lib-jitsi-meet';
 import { isFatalJitsiConnectionError } from './react/features/base/lib-jitsi-meet/functions';
+import { appendURLParam } from './react/features/base/util';
```
[the remark this answers is not in the document]
Comment (saghul) [reply connection.js:20]: Sorry, this one: https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/util/uri.js

[the remark this answers is not in the document]
Comment (damencho) [reply connection.js]: That is already there.

File: connection.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { LoginDialog } from './react/features/authentication/components';
 import { isTokenAuthEnabled } from './react/features/authentication/functions';
 import {
     connectionEstablished,
-    connectionFailed
+    connectionFailed, constructOptions
```
Comment (saghul) [inline connection.js:11]: style: put the import in a newline

File: react/features/app/actions.native.js [hunk 3]
```diff
@@ -15,8 +15,9 @@ import { connect, disconnect, setLocationURL } from '../base/connection';
 import { loadConfig } from '../base/lib-jitsi-meet/functions.native';
 import { createDesiredLocalTracks } from '../base/tracks';
 import {
+    appendURLParam,
     getBackendSafeRoomName,
-    parseURIString,
+    parseURIString, parseURLParams,
```
Comment (saghul) [inline react/features/app/actions.native.js:20]: style: put it in a new line

File: react/features/app/actions.web.js [hunk 4]
```diff
@@ -93,7 +94,7 @@ export function appNavigate(uri: ?string) {
         let url = `${baseURL}config.js`;
 
         // XXX In order to support multiple shards, tell the room to the deployment.
-        room && (url += `?room=${getBackendSafeRoomName(room)}`);
+        room && (url = appendURLParam(url, 'room', getBackendSafeRoomName(room)));
```
Comment (saghul) [inline react/features/app/actions.web.js:97]: I think it's a noop, by why didn't you add the release here?

File: react/features/base/connection/actions.any.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            bosh = `${locationURL.protocol}${bosh}`;
+        } else if (bosh.startsWith('/')) {
+            // Handle relative URLs, which won't work on mobile.
+            const {
+                protocol,
+                host,
+                contextRoot
+            } = parseURIString(locationURL.href);
+
+            // eslint-disable-next-line max-len
```
Comment (saghul) [inline react/features/base/connection/actions.any.js:49]: No longer needed I reckon.

File: react/features/base/connection/actions.any.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+} from '../util';
+
+import logger from './logger';
+
+/**
+ * Constructs options to be passed to the constructor of {@code JitsiConnection}
+ * based on the redux state.
+ *
+ * @param {Object} state - The redux state.
+ * @param {boolean} mobile - Whether we use it for mobile, whether to try enabling ab testing of websockets.
```
Comment (saghul) [inline react/features/base/connection/actions.any.js:16]: You don't need the parameter, you can check if `navigator.product === 'ReactNative'` to detect mobile without extra dependencies.

File: react/features/invite/functions.js [hunk 7]
```diff
@@ -612,6 +612,7 @@ export function getDialInfoPageURLForURIString(
     }
     const { protocol, host, contextRoot, room } = parseURIString(uri);
 
+    // TODO: do we care for releases here
```
Comment (saghul) [inline react/features/invite/functions.js:615]: Why not add it for consistency? We might need it one day?

**Review by saghul — CHANGES_REQUESTED** [review]
Left some comments, PTAL!

Question: what about the people search URLs? Does it make sense to pass the parameter around there?

File: react/features/app/actions.web.js (hunk 4 above)
Comment (damencho) [reply react/features/app/actions.web.js:97]: cause its noop :) but I'll add it to match the native code

File: connection.js (hunk 2 above)
Comment (damencho) [reply connection.js]: Fixed.

File: react/features/app/actions.native.js (hunk 3 above)
Comment (damencho) [reply react/features/app/actions.native.js]: Fixed.

File: react/features/base/connection/actions.any.js (hunk 5 above)
Comment (damencho) [reply react/features/base/connection/actions.any.js]: Dropped.

File: react/features/base/connection/actions.any.js (hunk 6 above)
Comment (damencho) [reply react/features/base/connection/actions.any.js]: Dropped the param.

File: react/features/invite/functions.js (hunk 7 above)
Comment (damencho) [reply react/features/invite/functions.js]: Done.

**Review by saghul — APPROVED** [review]

**Pull request merged**